### PR TITLE
Add build id to workflow context

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
* Add Build ID to workflow info
* Update Core

## Why?
Part of https://github.com/temporalio/features/issues/253

## Checklist

1. Closes 

2. How was this tested:
Added IT

3. Any docs updates needed?

### CODE REVIEW

File: src/Temporalio/Workflows/Workflow.cs [hunk 1]
```diff
@@ -37,6 +37,18 @@ public static class Workflow
...
+        /// Gets the current build id.
+        /// </summary>
+        /// <remarks>
+        /// This is the Build ID of the worker which executed the current Workflow Task. It may be
+        /// empty if the task was completed by a worker without a Build ID. If this worker is
+        /// the one executing this task for the first time and has a Build ID set, then its ID will
+        /// be used. This value may change over the lifetime of the workflow run, but is
+        /// deterministic and safe to use for branching.
+        /// </remarks>
+        public static string CurrentBuildId => Context.CurrentBuildId;
```
Comment (cretz) [inline src/Temporalio/Workflows/Workflow.cs:50]: Thank you for 1) not actually putting this on workflow info, and 2) respecting alphabetical order

File: src/Temporalio/Bridge/sdk-core
Comment (cretz) [inline src/Temporalio/Bridge/sdk-core:1]: You'll have to regen protos with the updated core (see https://github.com/temporalio/sdk-dotnet/#regenerating-protos, make sure it's 3.x `protoc` on the `PATH`). We have a CI check that makes sure our generated protos are up to date with the upstream API as seen in the core submodule and it looks like there have been additions.

File: tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs [hunk 2]
```diff
@@ -3873,14 +3939,28 @@ await foreach (var evt in handle.FetchHistoryEventsAsync())
         Func<TemporalWorker, Task> action,
         TemporalWorkerOptions? options = null,
         IWorkerClient? client = null)
+    {
+        await ExecuteWorkerAsyncReturning<TWf, bool>(
+            (w) =>
+            {
+                return action(w).ContinueWith(t => true);
+            },
```
Comment (cretz) [inline tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs]:
```suggestion
            w => action(w).ContinueWith(t => true),
```

But it's test code, so I don't really care

File: tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs [hunk 3]
```diff
@@ -1,3 +1,5 @@
+using System.Reflection.Metadata;
```
Comment (cretz) [inline tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs]: It wasn't obvious to me what is using this namespace

**Review by cretz — COMMENTED** [review]
LGTM, just need proto regen

File: src/Temporalio/Bridge/sdk-core
Comment (Sushisource) [reply src/Temporalio/Bridge/sdk-core:1]: Ugh right

File: tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs (hunk 3 above)
Comment (Sushisource) [reply tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs]: Probably the IDE did something silly and imported it

File: src/Temporalio/Bridge/sdk-core
Comment (Sushisource) [reply src/Temporalio/Bridge/sdk-core:1]: Done

**Review by cretz — APPROVED** [review]
Looks great, thanks!

**Pull request merged**